File: bindings/java/src/test/java/io/angzarr/router/conformance/saga/SagaFixture.java
package io.angzarr.router.conformance.saga;

import io.angzarr.router.conformance.Builders;
import io.angzarr.CommandBook;
import io.angzarr.Cover;
import io.angzarr.EventBook;
import io.angzarr.Notification;
import io.angzarr.RejectionNotification;
import io.angzarr.router.Destinations;
import io.angzarr.router.Thunks.SagaEmission;
import java.util.List;
import test.counter.Counter;
import test.counter.OrderSagaAngzarr;

/** The conformance OrderSaga fixture, implementing the generated seam: a
 * declared source event emits a Reserve command stamped with the supplied
 * destination sequence; a rejection injects one fact event. */
final class SagaFixture implements OrderSagaAngzarr.OrderSagaHandler {

  @Override
  public SagaEmission increased(Counter.Increased event, Destinations dests, Cover sourceCover) {
    CommandBook cmd = Builders.reserveCommand();
    if (dests.has("inventory")) {
      cmd = dests.stampCommand(cmd, "inventory");
    }
    return new SagaEmission(List.of(cmd), List.of());
  }

  @Override
  public List<EventBook> onReserveRejected(Notification n, RejectionNotification rejection) {
    return List.of(Builders.oneFact());
  }
}

File: bindings/csharp/conformance/Saga/SagaFixture.cs
using System;
using System.Collections.Generic;
using Angzarr;
using Angzarr.Router;
using TC = Test.Counter;

namespace Angzarr.Router.Conformance.Saga;

/// <summary>The conformance OrderSaga fixture: a declared source event emits a
/// Reserve command stamped with the supplied destination sequence; a rejection
/// injects one fact event.</summary>
internal sealed class SagaFixture : TC.OrderSagaAngzarr.OrderSagaHandler
{
    public SagaEmission Increased(TC.Increased ev, Destinations dests)
    {
        var cmd = Builders.ReserveCommand();
        if (dests.Has("inventory"))
        {
            cmd = dests.StampCommand(cmd, "inventory");
        }
        return new SagaEmission(new[] { cmd }, Array.Empty<EventBook>());
    }

    public IReadOnlyList<EventBook> OnReserveRejected(
        Notification n,
        RejectionNotification rejection
    ) => new[] { Builders.OneFact() };
}
